packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_prop/boolean_prop.rs

```diff
@@ -23,38 +23,69 @@ pub struct BooleanProp {
+    /// and we will fall back to `independent_state`, which will be initialized with this default.
     default_value: bool,
 
+    /// If `true`, then we will propagate `came_from_default` from the dependency
+    /// in the case where a single dependency is returned.
+    /// If `false`, then `came_from_default` will be true only if no dependencies were found
+    /// and we are returning an independent value that hasn't yet been changed from its default.
+    propagate_came_from_default: bool,
+
+    /// If `true`, then this prop was created from an attribute.
+    /// In this case, an empty string `""` is interpreted as `true`;
+    /// otherwise an empty string `""` is interpreted as `false`.
     from_attribute: bool,
 }
 
 /// The values of the dependencies created from the data queries
 #[add_dependency_data]
 #[derive(Debug, Default, PropDependencies, PropDataQueries)]
 pub struct RequiredData {
+    /// An independent state variable (that doesn't have any dependencies)
+    /// that is used if `propagate_came_from_default` is false
+    /// to store the value when there are no dependencies.
+    independent_state: PropView<bool>,
+
     /// A vector of the boolean or string values of the dependencies coming from the data_query
     booleans_and_strings: Vec<BooleanOrString>,
 }
 impl BooleanProp {
-    /// Creates a boolean prop that calculates its value from the given data query.
-    pub fn new(data_query: DataQuery) -> Self {
-        BooleanProp {
-            data_query,
-            ..Default::default()
-        }
-    }
-
     /// Creates a boolean prop that calculates its value from the component's children
     /// matching the `String` or `Boolean` profile.
     ///
     /// If there are no matching children, the prop will be initialized with `default_value`.
-    pub fn new_from_children(default_value: bool) -> Self {
+    ///
+    /// The `propagate_came_from_default` argument influences
+    /// the behavior of this prop's `came_from_default` flag when there is only one matching child
+    /// in the following manner:
+    /// - If `propagate_came_from_default` is `true`, then if there is only one matching child,
+    ///   `came_from_default` will match the `came_from_default` of that single child.
+    /// - If `propagate_came_from_default` is `false`, then `came_from_default`
+    ///   will always be `false` if there are any matching children.
+    ///
+    /// Regardless of the `propagate_came_from_default` argument,
+    /// - If there are two or more matching children, `came_from_default` is always `false`.
+    /// - If there are no matching children, then `came_from_default` will initially be true
+    ///   (and the value of the prop itself will be set to `default_value`).
+    ///   As soon as the value is changed (by a call to `invert()`),
+    ///   then `came_from_default` will be set to `false`.
+    pub fn new_from_children(default_value: bool, propagate_came_from_default: bool) -> Self {
```

Comment:
Not thrilled about this extra param. Seems like you almost always set it as `true`?

I suggest leaving it off of the `new_from_children` and creating an additional `new_from_children_no_propagate_came_from_default`. Let the ugly name be used in the few cases its needed.

Reply:
Right, I wasn't sure whether to duplicate code and make a separate function or add another parameter. I guess could make the current function be a non-public one that is called from `new_from_children` and `new_from_children_no_propagate_came_from_default`